pages/index.js

```diff
@@ -1,55 +1,36 @@
 import Head from 'next/head';
-import { useState } from 'react';
-import Articles from './components/Articles';
-import Sources from './components/Sources';
+import Articles from '../components/Articles';
+import Sources from '../components/Sources';
+import NewsContextProvider from '../lib/news-context';
 
-export const getStaticProps = async () => {
-  const res = await fetch(
-    "https://newsapi.org/v2/everything?q=apple&from=2021-09-21&to=2021-09-21&sortBy=popularity&apiKey=0cb60059c1974e85a3b1b8ef1d36036b"
-  );
-  const data = await res.json();
+export default function Home({ sources }) {
 
-  return {props: {data}}
+  return (
+    <NewsContextProvider>
+      <div className="container">
+        <div className="row">
+          <div className="col-lg-4">
+            <Sources {...{ sources }} />
+          </div>
+
+          <div className="col-lg-8">
+            <Articles />
+          </div>
+        </div>
+      </div>
+    </NewsContextProvider>
+  )
 }
 
 
-export default function Home({ data }) {
-
-  
-  const [articles, setArticles] = useState(data.articles);
-
-  //getting the unique values
-  const sortedSources = ["All", ...new Set(articles.map((item) => item.source.name)),];
+export const getStaticProps = async () => {
+  const baseUrl = process.env.NEXT_PUBLIC_API_URL
+  const apiKey = process.env.NEXT_PUBLIC_API_KEY
+  const res = await fetch(`${baseUrl}/top-headlines/sources?country=us&apiKey=${apiKey}`);
```

Comment:
My expectation was that you fetch the sources from `us` and allow the user to select from the list of sources. So that's what I'm doing here. Instead of fetching all articles and providing a filter with the sources included.